src/vm/perfmap.cpp

```diff
@@ -218,6 +222,38 @@ void PerfMap::LogJITCompiledMethod(MethodDesc * pMethod, PCODE pCode, size_t cod
     }
 }
 
+// Log a set of stub to the map.
+void PerfMap::LogStubs(const char* stubType, const char* stubOwner, PCODE pCode, size_t codeSize)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    if (s_Current == nullptr)
+    {
+        return;
+    }
+
+    // Logging failures should not cause any exceptions to flow upstream.
+    EX_TRY
+    {
+        if(!stubOwner)
+        {
+            stubOwner = "?";
+        }
+        if(!stubType)
+        {
+            stubOwner = "?";
+        }
+
+        // Build the map file line.
+        SString line;
+        line.Printf("%p %x stub<%d> %s<%s>\n", pCode, codeSize, ++(s_Current->m_StubsMapped), stubType, stubOwner);
```

Comment:
Do these stub numbers map back to anything?  Not sure if there are better numbers or names that we can map them back to, but it would be great to be able to look at these and map them back to things in the process.  If there's no good names, what about using the address of the stub instead of the unique number here?